oak_restricted_kernel/src/payload.rs

```diff
@@ -15,34 +15,55 @@
 //
 
 use crate::syscall::mmap::mmap;
+use alloc::{vec, vec::Vec};
+use anyhow::Result;
 use core::arch::asm;
-use goblin::elf64::program_header::{ProgramHeader, PF_W, PF_X, PT_LOAD};
+use goblin::elf64::program_header::{PF_W, PF_X, PT_LOAD};
+use oak_channel::Channel;
 use oak_restricted_kernel_interface::syscalls::{MmapFlags, MmapProtection};
 use x86_64::{
     structures::paging::{PageSize, Size2MiB},
     VirtAddr,
 };
 
+/// Reads a chunk of data and acknowledges the transmission by writing back the number of bytes
+/// read.
+fn read_chunk(channel: &mut dyn Channel, chunk: &mut [u8]) -> Result<()> {
```

Comment:
Would it really be too much to use the same [framing](https://github.com/project-oak/oak/blob/a39f5a659ce8c512dcf56c7ad0a6b0b0ab2d4df7/oak_channel/src/frame.rs) mechanism we use in the Oak Restricted Kernel? I feel that would solve this a bit more nicely. It does add a dep on micro_rpc though, while in principle it should be possible to split it to a crate that doesn't, if necessary. But since you are already thinking about the control channel, may as well take the micro_rpc dep now, assuming it works.

Up to you anyways, maybe @conradgrobler has opinions too.

Reply:
Yeah, I really want to avoid pulling in micro_rpc, if at all possible. There's no RPCs here, just a blind dump of data. Even in the future (wrt control channel), I would want to avoid micro_rpc in the kernel -- the kernel should really treat the channel as a dumb pipe.

Reply:
I think this approach is a bit clunky and brittle, but I think it is fine for now. I agree that splitting the framing code so that it does not automatically bring in micro_rpc might be a good option. It might also be an option to load the binary via a different mechanism (e.g. the QEMU fw_cfg device), so I think we should do some more design work before changing this.